scene/gui/dialogs.cpp

```diff
@@ -286,18 +286,29 @@ void AcceptDialog::_custom_action(const String &p_action) {
 	custom_action(p_action);
 }
 
+void AcceptDialog::_custom_button_visibility_changed(Button *button) {
+	Control *right_spacer = Object::cast_to<Control>(button->get_meta("__right_spacer"));
+	if (right_spacer) {
+		right_spacer->set_visible(button->is_visible());
+	}
+}
+
 Button *AcceptDialog::add_button(const String &p_text, bool p_right, const String &p_action) {
 	Button *button = memnew(Button);
 	button->set_text(p_text);
 
+	Control *right_spacer;
 	if (p_right) {
 		buttons_hbox->add_child(button);
-		buttons_hbox->add_spacer();
+		right_spacer = buttons_hbox->add_spacer();
 	} else {
 		buttons_hbox->add_child(button);
 		buttons_hbox->move_child(button, 0);
-		buttons_hbox->add_spacer(true);
+		right_spacer = buttons_hbox->add_spacer(true);
 	}
+	button->set_meta("__right_spacer", right_spacer);
+
+	button->connect("visibility_changed", callable_mp(this, &AcceptDialog::_custom_button_visibility_changed).bind(button));
```

Comment:
Since we're disconnecting other signals when the button is removed, we should probably disconnect this one as well.

Reply:
It should be auto disconnected when the spacer is freed though.

`remove_button` doesn't free the button itself and hence the other signals need to be manually disconnected.

Reply:
This line has no spacer in it :) It's a signal from a button and we connect it to the dialog.

Reply:
:man_facepalming: